### PR TITLE
Replace XAudio2 implementation with an unified Xaudio2Redist

### PR DESCRIPTION
**Pull request by CookiePLMonster**
This removes dual implementation for 2.7 and 2.8/2.9 interfaces
and also removes reliance on DirectX End User Runtimes for Windows 7.

### CODE REVIEW

File: 3rdparty/XAudio2Redist/include/xaudio2redist.h [hunk 1]
```diff
@@ -7,11 +7,31 @@
  *
  **************************************************************************/
 
+#ifdef _MSC_VER
 #pragma once
+#endif
```
Comment (MSuih) [inline 3rdparty/XAudio2Redist/include/xaudio2redist.h]: Why? Pragma once works fine with all compilers.

**Comment by MSuih** [comment]
You need to edit cmake files as well or otherwise this won't compile on that. 

Also, consider addressing #6723 as well while you're changing xaudio.

**Comment by CookiePLMonster** [comment]
> Also, consider addressing #6723 as well while you're changing xaudio.

Good call, I will - I unironically like working with XAudio2 so it'll be neat.

**Comment by CookiePLMonster** [comment]
Updated with CMake fixes (hopefully) and also made it handle no output devices gracefully. I made `AudioBackend` able to report "failure" in creation in which case a null handler gets used.

Might be useful for other audio backends too, but I didn't touch them atm.

**Comment by Megamouse** [comment]
The QtMsBuild PR will take a while (possibly waiting for appveyor to fix the windows images) so there is no need to wait in that regard.

**Comment by huteri** [comment]
this actually causing another bug https://github.com/RPCS3/rpcs3/issues/7678

**Comment by AniLeo** [comment]
No, you were never able to change those kind of settings mid game

**Pull request merged**